storage/oasis/nodeapi/file/kvstore.go

```diff
@@ -157,7 +157,7 @@ func fetchTypedValue[Value any](cache KVStore, key CacheKey, value *Value) error
 	}
 	err = cbor.Unmarshal(raw, value)
 	if err != nil {
-		return fmt.Errorf("failed to unmarshal key %s from cache into %T: %v", key.Pretty(), value, err)
+		return fmt.Errorf("failed to unmarshal key %s and value %x from cache into %T: %v", key.Pretty(), raw, value, err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to unmarshal the value for key %s from cache into %T: %v; raw value was %x", key.Pretty(), value, err, raw)
```

For large values, this keeps the more important part of the err msg together and readable.